### PR TITLE
Loopback API

### LINKED ISSUE #63: Issue: dfi_buffer_write_packets doesn't check for empty list
### Issue
`dfi_buffer_write_packets` function assumes that list isn't empty. This can lead to a NULL pointer issue further down the stack.

### LINKED ISSUE #64: Feature Request: Loopback API support
### Feature Request
WDDR should have an API to put the PHY into EXTERNAL_LOOPBACK mode.

### PR DESCRIPTION
**Pull request by wavious-wpatty**
Fixes:
  - Fixes #63 (Empty packet list issue)
  - Fixes issue where Frequency Switch FSM handles an
    invalid init_start interrupt

Features:
  - Driver API to override for init_start and init_complete
    now allows for override value to be set
  - Loopback API support (#64)

### CODE REVIEW

File: dev/wddr/device.c [hunk 1]
```diff
@@ -587,6 +591,65 @@ static void wddr_iocal_calibrate(void *dev)
     zqcal_calibrate(&wddr->cmn.zqcal, &wddr->table->cal.common.common.zqcal);
 }
 
+static void ca_enable_loopback(ca_path_t *ca_path, driver_cfg_t *cfg)
+{
+    driver_override_all_bits(&ca_path->tx.ca.driver, WDDR_SLICE_TYPE_CA, cfg, false);
```
Comment (wavious-jbasista) [inline dev/wddr/device.c:596]: Just as a curiousity, it is correct that override is false right?

File: drivers/dfi/driver.c [hunk 2]
```diff
@@ -128,25 +128,37 @@ void dfi_ovr_traffic_cfg_reg_if(dfi_dev_t *dfi, dfi_ovr_traffic_cfg_t *cfg)
...
     *init_start = GET_REG_FIELD(reg_read(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_STA__ADR), DDR_DFI_STATUS_IF_STA_REQ);
 }
+
+void dfi_ca_read_loopback_enable_reg_if(bool enable)
+{
+    uint32_t reg_val;
+    reg_val = reg_read(WDDR_MEMORY_MAP_DFI_CH0 +  DDR_DFICH_TOP_1_CFG__ADR);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFICH_TOP_1_CFG_CA_RDDATA_EN, enable);
+    reg_write(WDDR_MEMORY_MAP_DFI_CH0 +  DDR_DFICH_TOP_1_CFG__ADR, reg_val);
+}
```
Comment (wavious-jbasista) [inline drivers/dfi/driver.c:164]: This only supports CH0 correct?  Is loopback supported on CH1?

File: include/dev/driver/device.h [hunk 3]
```diff
@@ -113,4 +113,21 @@ void driver_override(driver_dev_t *driver,
...
                      bool override);
 
+/**
+ * @brief   Driver Override All Bits
+ *
+ * @details Sets the Driver device settings for all bits for the given slice.
+ *
+ * @param[in]   driver      pointer to Driver device.
+ * @param[in]   slice_type  slice type of Driver device.
+ * @param[in]   cfg         configuraton used to configure driver.
```
Comment (wavious-jbasista) [inline include/dev/driver/device.h]: lets identify this as a pointer

**Review by wavious-jbasista — CHANGES_REQUESTED** [review]

File: dev/wddr/device.c (hunk 1 above)
Comment (wavious-wpatty) [reply dev/wddr/device.c:596]: Yes, it is. But the variable isn't correct. It should be output enable. I will update.

File: drivers/dfi/driver.c (hunk 2 above)
Comment (wavious-wpatty) [reply drivers/dfi/driver.c:164]: I will add Channel 1.

Comment (wavious-wpatty) [reply drivers/dfi/driver.c:164]: Actually, it's not supported. There's no DFI CH1 in the memory map. This is because of 1x32 configuration.

**Review by wavious-jbasista — APPROVED** [review]
Looks good!

**Pull request merged**